api/v1beta1/vcdcluster_conversion.go

```diff
@@ -1,5 +1,30 @@
 package v1beta1
 
-func (*VCDCluster) Hub() {}
+import (
+	"github.com/vmware/cluster-api-provider-cloud-director/api/v1beta2"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
 
-func (*VCDClusterList) Hub() {}
+// ConvertTo converts this VCDCluster to the Hub version (v1beta2).
+func (src *VCDCluster) ConvertTo(dstRaw conversion.Hub) error {
+	dst := dstRaw.(*v1beta2.VCDCluster)
+	return Convert_v1beta1_VCDCluster_To_v1beta2_VCDCluster(src, dst, nil)
```

Comment:
This needs to have the logic to restore fields using `utilconversion.UnmarshalData(src, restored)`